crates/merkle-inclusion/src/lib.rs

```diff
@@ -0,0 +1,158 @@
+pub mod traits;
+
+use crate::traits::GadgetDigest;
+use bellpepper_core::boolean::Boolean;
+use bellpepper_core::{ConstraintSystem, SynthesisError};
+use ff::PrimeField;
+
+// HashValue represents the digest type output from the hash function
+pub type HashValue = Vec<Boolean>;
+pub type Key = Vec<Boolean>;
+const HASH_LENGTH: usize = 32 * 8;
+
+/// Represents a leaf node in a Jellyfish Merkle Tree.
+///
+/// Each leaf node contains a key-value pair, where the key is used to determine the position in the tree and the value
+/// is the data stored.
+pub struct Leaf {
+    key: Key,
```

Comment:
What's the expected endianness of this key? Shall we document it?

Reply:
As of now I handle the key to be Little Endian all the time. I think it would actually make more sense if it could be of the same endian of the hashing algorithm it is being associated to. Will update the code to reflect it.

Reply:
Did this in 775a5fa

Reply:
Behavior needed correction, updated in a74aa26